### PR TITLE
[4.0] Add ordering column in com_banners

### PR DESCRIPTION
**Pull request by hardik-codes**
This PR adds filtering column which was missing in com_banners(clients) and makes it consistent with other components

### After Patch
com_banners

### Documentation Changes Required
None

### CODE REVIEW

File: administrator/components/com_banners/tmpl/clients/default.php [hunk 1]
```diff
@@ -103,6 +105,26 @@
...
+										if (!$canChange)
+										{
+										$iconClass = ' inactive';
+										}
+										elseif (!$saveOrder)
+										{
+										$iconClass = ' inactive tip-top hasTooltip" title="' . HTMLHelper::_('tooltipText', 'JORDERINGDISABLED');
+										}
+										?>
+										<span class="sortable-handler<?php echo $iconClass ?>">
```
Comment (Quy) [inline administrator/components/com_banners/tmpl/clients/default.php]:
```suggestion
										<span class="sortable-handler<?php echo $iconClass; ?>">
```

File: administrator/components/com_banners/tmpl/clients/default.php [hunk 2]
```diff
@@ -103,6 +105,26 @@
 								$canChange  = $user->authorise('core.edit.state', 'com_banners') && $canCheckin;
 								?>
 								<tr class="row<?php echo $i % 2; ?>">
+                                    					<td class="order text-center d-none d-md-table-cell">
+										<?php
+										$iconClass = '';
+
+										if (!$canChange)
+										{
+										$iconClass = ' inactive';
```
Comment (Quy) [inline administrator/components/com_banners/tmpl/clients/default.php]:
```suggestion
											$iconClass = ' inactive';
```

File: administrator/components/com_banners/tmpl/clients/default.php [hunk 3]
```diff
@@ -103,6 +105,26 @@
...
+										}
+										elseif (!$saveOrder)
+										{
+										$iconClass = ' inactive tip-top hasTooltip" title="' . HTMLHelper::_('tooltipText', 'JORDERINGDISABLED');
```
Comment (Quy) [inline administrator/components/com_banners/tmpl/clients/default.php]:
```suggestion
											$iconClass = ' inactive tip-top hasTooltip" title="' . HTMLHelper::_('tooltipText', 'JORDERINGDISABLED');
```

**Comment by brianteeman** [comment]
Sorry but this is completely wrong.

I am assuming from the code that you have added "ordering" not "filtering"


First of all this patch gives two php notices
```
 
Notice: Undefined variable: saveOrder in C:\htdocs\joomla-cms\administrator\components\com_banners\tmpl\clients\default.php on line 116
 
Notice: Undefined variable: saveOrder in C:\htdocs\joomla-cms\administrator\components\com_banners\tmpl\clients\default.php on line 124
```

It probably does that because there is no column in the database to store the ordering.

Even if you fix both of those then the main question is "why"

There is no reason to add the ability to set an order for clients. What would it allow you to do with banners that you cannot do now.

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on fe68691c66fd844ac7c44e70a905ecfb017f1026<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24646">issues.joomla.org/tracker/joomla-cms/24646</a>.</sub>

**Comment by hardik-codes** [comment]
> There is no reason to add the ability to set an order for clients. What would it allow you to do with banners that you cannot do now.

@brianteeman I did this to make it consistent with other components

**Comment by brianteeman** [comment]
This still does nothing at all because there is no column in the database for ordering.

I am all for consistency but in this case it is wrong. There is no point in ordering the clients as it doesn't have any function. Only add code/features if they do something. We aslo dont have the ability to order users for the same reason - it serves no purpose

**Comment by richard67** [comment]
I have tested this item :red_circle: unsuccessfully on eb993bfe544b8724fd4aca7ab7b74357223e57fe<br /><br />This PR is wrong for reasons stated by @brianteeman above:
- The banner clients table in database does not have a column for ordering, in opposite to the banners, where it makes sense.
- To add ordering for banner clients (including column in the banner_clients database table) does not provide any useful functionality and so does not make sense.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24646">issues.joomla.org/tracker/joomla-cms/24646</a>.</sub>

**Comment by hardik-codes** [comment]
Closing for reasons stated above
Thanks @brianteeman @richard67 for the info

**Comment by hardik-codes** [comment]
@brianteeman in the following link ```administrator/index.php?option=com_menus&view=items&menutype=``` also there is no ordering column.
Is an ordering column needed here?

**Comment by brianteeman** [comment]
Will it to anything? Is there an ordering column?

**Comment by hardik-codes** [comment]
As of now there is no ordering column

**Comment by richard67** [comment]
I think Brian means if there is such column in the database table for this view.

**Comment by hardik-codes** [comment]
Ok 
How do I check this @richard67 ?

**Comment by richard67** [comment]
@hardik-codes With phpmyadmin or pgmyadmin, dependent on your type of database. Table is `#__menu`, replace `#__` by your DB prefix.

But there is no need to check that. In case of menu items, ordering is a bit different.

The ordering of menu items makes only sense within one menu type. That's why the "all menu items" view does not show an ordering column, but the "menu items for a particular menu type" view shows such a column.

I think I remember there was a PR in past which made this change for the "all menu items" view.

**Pull request closed without merging**